pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -824,6 +829,130 @@ impl DeltaLayerInner {
         }
     }
 
+    // Look up the keys in the provided keyspace and update
+    // the reconstruct state with whatever is found.
+    //
+    // If the key is cached, go no further than the cached Lsn.
```

Comment:
Again, `cached_lsn`, we can probably get rid of that.

Reply:
We can get rid of `cached_lsn`, but the search should still allow for using a materialised page to stop early. As we discussed, we are not going to do this for vectored searches, but we might still want to do it for single gets which are proxied into the vectored impl in order to limit the number of things changing when we doing the switch over in the future.